File: frontend/src/api/logistics.ts
import client from './client';

export interface LogisticsSummary {
  activeOrders: number;
  inTransit: number;
  deliveredToday: number;
  exceptionOrders: number;
  activeRoutes: number;
  onTimeRate: number;
}

export interface LogisticsOverview {
  generatedAtUtc: string;
  summary: LogisticsSummary;
  alerts: Array<{
    orderNumber: string;
    customerName: string;
    routeCode: string;
    status: string;
    exceptionReason: string;
    attemptCount: number;
    riderName: string;
    etaUtc: string;
  }>;
  routeCards: LogisticsRoute[];
  orderCards: LogisticsOrder[];
  liveStops: LogisticsStop[];
}

export interface LogisticsOrder {
  id: string;
  orderNumber: string;
  customerName: string;
  customerSegment: string;
  salesChannel: string;
  city: string;
  area: string;
  status: string;
  priority: string;
  itemCount: number;
  orderValue: number;
  routeCode: string;
  driverName: string;
  vehicleNumber: string;
  dispatchNotes: string;
  createdAtUtc: string;
  promisedAtUtc?: string;
  dispatchedAtUtc?: string;
  deliveredAtUtc?: string;
}

export interface LogisticsRoute {
  id: string;
  routeCode: string;
  hub: string;
  territory: string;
  driverName: string;
  vehicleNumber: string;
  status: string;
  plannedStops: number;
  completedStops: number;
  distanceKm: number;
  completionPercent: number;
  currentStop: string;
  nextStop: string;
  plannedForDate: string;
  departureTimeUtc: string;
  etaCompleteUtc?: string;
  notes: string;
}

export interface LogisticsStop {
  id: string;
  orderNumber: string;
  routeCode: string;
  customerName: string;
  addressLine: string;
  city: string;
  status: string;
  proofStatus: string;
  recipientName: string;
  attemptCount: number;
  riderName: string;
  timeWindow: string;
  etaUtc: string;
  deliveredAtUtc?: string;
  exceptionReason: string;
}

export const logisticsApi = {
  overview: () =>
    client.get<LogisticsOverview>('/api/logistics/overview').then((r) => r.data),

  orders: (params: { status?: string; page?: number; pageSize?: number } = {}) =>
    client.get<{ total: number; page: number; pageSize: number; items: LogisticsOrder[] }>('/api/logistics/orders', { params }).then((r) => r.data),

  routes: (params: { status?: string } = {}) =>
    client.get<{ items: LogisticsRoute[] }>('/api/logistics/routes', { params }).then((r) => r.data),

  lastMile: (params: { status?: string; page?: number; pageSize?: number } = {}) =>
    client.get<{ total: number; page: number; pageSize: number; items: LogisticsStop[] }>('/api/logistics/last-mile', { params }).then((r) => r.data),

  dispatchOrder: (id: string, body: { routeCode?: string; driverName?: string; vehicleNumber?: string; notes?: string }) =>
    client.post<LogisticsOrder>(`/api/logistics/orders/${id}/dispatch`, body).then((r) => r.data),

  progressRoute: (id: string, body: { completedStopsDelta?: number; currentStop?: string; nextStop?: string; etaCompleteUtc?: string; notes?: string }) =>
    client.post<LogisticsRoute>(`/api/logistics/routes/${id}/progress`, body).then((r) => r.data),

  confirmDelivery: (id: string, body: { recipientName?: string; proofStatus?: string; exceptionReason?: string }) =>
    client.post<LogisticsStop>(`/api/logistics/stops/${id}/deliver`, body).then((r) => r.data),
};

File: backend-dotnet/Zayra.Api/Models/Logistics.cs
using Zayra.Api.Domain.Entities;

namespace Zayra.Api.Models;

public class DispatchOrder : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public string CustomerSegment { get; set; } = string.Empty;
    public string SalesChannel { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Area { get; set; } = string.Empty;
    public string Status { get; set; } = "Queued"; // Queued/Picking/Packed/Dispatched/InTransit/Delivered/Exception/Returned
    public string Priority { get; set; } = "Normal"; // Low/Normal/High/Critical
    public int ItemCount { get; set; }
    public decimal OrderValue { get; set; }
    public string RouteCode { get; set; } = string.Empty;
    public string DriverName { get; set; } = string.Empty;
    public string VehicleNumber { get; set; } = string.Empty;
    public string DispatchNotes { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? PromisedAtUtc { get; set; }
    public DateTime? DispatchedAtUtc { get; set; }
    public DateTime? DeliveredAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

public class DeliveryRoute : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string RouteCode { get; set; } = string.Empty;
    public string Hub { get; set; } = string.Empty;
    public string Territory { get; set; } = string.Empty;
    public string DriverName { get; set; } = string.Empty;
    public string VehicleNumber { get; set; } = string.Empty;
    public string Status { get; set; } = "Planned"; // Planned/Ready/Active/Delayed/Closed
    public int PlannedStops { get; set; }
    public int CompletedStops { get; set; }
    public decimal DistanceKm { get; set; }
    public decimal CompletionPercent { get; set; }
    public string CurrentStop { get; set; } = string.Empty;
    public string NextStop { get; set; } = string.Empty;
    public DateTime PlannedForDate { get; set; } = DateTime.UtcNow.Date;
    public DateTime DepartureTimeUtc { get; set; } = DateTime.UtcNow;
    public DateTime? EtaCompleteUtc { get; set; }
    public string Notes { get; set; } = string.Empty;
}

public class LastMileStop : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string RouteCode { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public string AddressLine { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Status { get; set; } = "OutForDelivery"; // OutForDelivery/Attempted/Delivered/Rescheduled/Failed
    public string ProofStatus { get; set; } = "None"; // None/POD/OTP/Signature
    public string RecipientName { get; set; } = string.Empty;
    public int AttemptCount { get; set; }
    public string RiderName { get; set; } = string.Empty;
    public string TimeWindow { get; set; } = string.Empty;
    public DateTime EtaUtc { get; set; } = DateTime.UtcNow;
    public DateTime? DeliveredAtUtc { get; set; }
    public string ExceptionReason { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAtUtc { get; set; }
}
